File: tempcode1.py
def is_prime(n):
  """
  Checks if a number is prime.

  Args:
    n: The number to check.

  Returns:
    True if n is prime, False otherwise.
  """
  if n <= 1:
    return False
  for i in range(2, int(n**0.5) + 1):
    if n % i == 0:
      return False
  return True

def find_primes_in_string(s):
  """
  Finds and prints all prime numbers within a string.

  Args:
    s: The input string.
  """
  primes = []
  for word in s.split():
    try:
      num = int(word)
      if is_prime(num):
        primes.append(num)
    except ValueError:
      pass  # Skip non-numeric words

  if primes:
    print("Prime numbers found in the string:")
    for prime in primes:
      print(prime)
  else:
    print("No prime numbers found in the string.")

# Interactive part
while True:
  try:
    n = int(input("Enter a number to check for primality (or 'q' to quit): "))
    if n == 'q':
      break
    if is_prime(n):
      print(f"{n} is a prime number.")
    else:
      print(f"{n} is not a prime number.")

    s = input("Enter a string to find prime numbers in: ")
    find_primes_in_string(s)
  except ValueError:
    print("Invalid input. Please enter an integer or 'q' to quit.")

File: tempcode1.java
import java.util.regex.*;
import java.util.ArrayList;
import java.util.List;

public class PrimeChecker {

    // Function to check if a number is prime
    public static boolean isPrime(int n) {
        if (n <= 1) return false;
        else if (n == 2) return true;
        else if (n % 2 == 0) return false;

        for (int i = 3; i * i <= n; i += 2) {
            if (n % i == 0) return false;
        }
        return true;
    }

    // Function to find all primes in a string
    public static List<Integer> findPrimesInString(String input) {
        Pattern pattern = Pattern.compile("\\d+");
        Matcher matcher = pattern.matcher(input);

        List<Integer> primes = new ArrayList<>();
        while (matcher.find()) {
            int num = Integer.parseInt(matcher.group());
            if (isPrime(num)) {
                primes.add(num);
            }
        }
        return primes;
    }

    public static void main(String[] args) {
        System.out.println("Enter a string:");
        Scanner scanner = new Scanner(System.in);
        String inputString = scanner.nextLine();

        List<Integer> primeNumbers = findPrimesInString(inputString);

        System.out.println("Prime numbers found in the string: " + primeNumbers);
    }
}